sdk/keyvault/azkeys/client.go

```diff
@@ -284,15 +284,35 @@ type CreateRSAKeyOptions struct {
 
 	// Application specific metadata in the form of key-value pairs.
 	Tags map[string]string `json:"tags,omitempty"`
+
+	// Elliptic curve name. For valid values, see JsonWebKeyCurveName.
+	Curve *JSONWebKeyCurveName `json:"crv,omitempty"`
+
+	// The attributes of a key managed by the key vault service.
+	KeyAttributes *KeyAttributes         `json:"attributes,omitempty"`
+	KeyOps        []*JSONWebKeyOperation `json:"key_ops,omitempty"`
+
+	// The policy rules under which the key can be exported.
+	ReleasePolicy *KeyReleasePolicy `json:"release_policy,omitempty"`
 }
 
 // convert CreateRSAKeyOptions to generated.KeyCreateParameters
 func (c CreateRSAKeyOptions) toKeyCreateParameters(k KeyType) generated.KeyCreateParameters {
+	var keyOps []*generated.JSONWebKeyOperation
```

Comment:
It's not a big deal in this case but you can make this code more efficient by pre-allocating the entire array:
var keyOps []*generated.JSONWebKeyOperation = make([]*generated.JSONWebKeyOperation, 0, len(c.KeyOps))

This is a great pattern to know and use whenever possible as it reduces allocations and improves performance by reducing GCs and by avoiding copying when append needs to grow the underlying array.

Reply:
I don't use a `make` statement because if `c.KeyOps` is nil, we don't want to send an empty struct in the body of the request. This bypasses that. I can use `make` instead of append inside of the `if` statement